### PR TITLE
[+] [Fix] [Bug] Dollar sign causing rendering issues

### PR DESCRIPTION
**Pull request by H0llyW00dzZ**
Ref : [Bug] Dollar sign causing rendering issues? #2841

### CODE REVIEW

**Comment by H0llyW00dzZ** [comment]
**Note:**
For other developers, it is recommended to use `<Markdown content={mdText} />` instead. If you use `<Markdown content={mdText} onClick={copy} />`, it will copy all the text in the markdown. This issue with the dollar sign has been fixed in the recent commits.

File: app/components/markdown.tsx [hunk 1]
```diff
@@ -100,15 +111,26 @@ export function PreCode(props: { children: any }) {
...
+    /(`{3}[\s\S]*?`{3}|`[^`]*`)|\$/g,
+    (match, codeBlock) => {
+      // Exclude code blocks from replacement
+      if (codeBlock) {
+        return match; // Return the code block as it is
+      } else {
+        return "&#36;"; // Escape dollar signs outside of code blocks
+      }
+    },
+  );
```
Comment (H0llyW00dzZ) [inline app/components/markdown.tsx:125]: Note:
not yet complete; still need filter each instance individually for a real dollar sign, such as in code blocks.

**Comment by a6z6** [comment]
@H0llyW00dzZ , Great, I tried to modify your code, works fine! 🙌🏻

```ts
function _MarkDownContent(props: { content: string }) {
  const userAgent = window.navigator.userAgent;
  const isAppleIosDevice = isIOS(); // Load isAppleIosDevice from isIOS functions

  // According to this post: https://www.drupal.org/project/next_webform/issues/3358901
  // iOS 16.4 is the first version to support lookbehind
  const iosVersionSupportsLookBehind = 16.4;
  let doesIosSupportLookBehind = false;

  if (isAppleIosDevice) {
    const match = /OS (\d+([_.]\d+)+)/.exec(userAgent);
    if (match && match[1]) {
      const iosVersion = parseFloat(match[1].replace("_", "."));
      doesIosSupportLookBehind = iosVersion >= iosVersionSupportsLookBehind;
    }
  }

  let escapedContent = props.content;
  if (isAppleIosDevice && !doesIosSupportLookBehind) {
    escapedContent = props.content.replace(
      // Exclude code blocks & math block from replacement
      // custom-regex for unsupported Apple devices
      /(`{3}[\s\S]*?`{3}|`[^`]*`)|(\$(?!\$))/g,
      (match, codeBlock) => {
        if (codeBlock) {
          return match; // Return the code block as it is
        } else {
          return "&#36;"; // Escape dollar signs outside of code blocks
        }
      },
    );
  } else {
    escapedContent = props.content.replace(
      // Exclude code blocks & math block from replacement
      /(`{3}[\s\S]*?`{3}|`[^`]*`)|(?<!\$)(\$(?!\$))/g,
      (match, codeBlock) => {
        if (codeBlock) {
          return match; // Return the code block as it is
        } else {
          return "&#36;"; // Escape dollar signs outside of code blocks
        }
      },
    );
  }

  return (

...

```

### The actual testing result shows, iOS 15.4, iOS 16.1, iOS 16.6, Chrome Browser, all works well with dollar sign! 💰

Hope this helps!

**Comment by H0llyW00dzZ** [comment]
@a6z6 nice I'll update the commits with ur code

**Comment by H0llyW00dzZ** [comment]
https://chatgpt.btz.sh/#/changelog

**Comment by Yidadaa** [comment]
Great work! It could be better to extract the logic to a new function `escapeMarkdownContent(content: string): string`, and use it in the component with `const escapedContent = useMemo(() => ...)`.

**Comment by H0llyW00dzZ** [comment]
@Yidadaa alright I noted that 
this issue would be fix in v2.9.9 later

**Comment by H0llyW00dzZ** [comment]
@Yidadaa check

**Comment by Yidadaa** [comment]
image

Looks like a fail on this simple case, here's the output on GitHub:

---

what is $e=mc^2$ ?

---

image

**Comment by H0llyW00dzZ** [comment]
you have to add 5 backticks 

image

**Comment by Yidadaa** [comment]
Why? What syntax is this? I have never seen this syntax for expressing mathematical formulas.

**Comment by H0llyW00dzZ** [comment]
@Yidadaa its this one $e=mc^2$ but it not possible I think for trying fix this formula `$e=mc^2$`  (only this)  in preview/user unless it response from ai.
and other formula should be work normally

**Comment by Yidadaa** [comment]
@H0llyW00dzZ My question is why should we "add 4 backticks" to it? Is it a workaround?

**Comment by H0llyW00dzZ** [comment]
its default from package markdown I think 
because in this pr $ it replaced by `"&#36;"`

image

**Comment by Yidadaa** [comment]
I understand what you mean. In most cases, chatgpt will try to wrap latex formulas in code block syntax. Your solution tries to parse the formulas in the code block, but this is not a commonly used markdown syntax. 

Normal markdown The syntax is to use the $ symbol to wrap the formula directly in the text. We should learn how github handles formula input and only replace the text with blank characters before and after $.

**Comment by H0llyW00dzZ** [comment]
yeah but its bug for dollar sign in markdown

**Comment by Yidadaa** [comment]
#3237

**Pull request closed without merging**